src/hooks/use-web3-call-state.ts

```diff
@@ -187,3 +204,174 @@ export function useWeb3CallState(
     joinCall,
   };
 }
+
+export function useWeb3SolCallState(
```

Comment:
nit: It seems that a lot of this logic is copied from the `useWeb3CallState` function above. What's the reason for reimplementing it on this branching logic rather than branching inside the function we've already declared?